src/Sentry/Protocol/Transaction.cs

```diff
@@ -155,27 +195,32 @@ internal Transaction(IHub hub)
         public void SetTag(string key, string value) =>
             (_tags ??= new Dictionary<string, string>())[key] = value;
 
-        /// <inheritdoc />
-        public ISpan StartChild(string operation)
+        internal ISpan StartChild(SpanId parentSpanId, string operation)
         {
-            var span = new Span(_spanRecorder, null, SpanId, operation);
-            _spanRecorder.Add(span);
+            var span = new Span(this, parentSpanId, operation)
+            {
+                IsSampled = IsSampled
+            };
+
+            (_spans ??= new List<Span>()).Add(span);
```

Comment:
Since `StartChild` can be called by multiple threads, this could be a problem

Reply:
can we make the list of spans an eagerly initialized `ConcurrentBag`? really don't want to deal with `Lazy<T>` or locks here.

Reply:
Lets leave it out so far, we might just pull Transaction out of the Scope in favor of a list of Spans (that could then be concurrent). Lets discuss this later